Week 1 683380312-6 นางสาววิลาสินี รอดจันทร์พะเนา 

Lab 1.1: Environment Setup & First Script

In [1]:
print("Hello World")
print("Welcome toScript Programming 101!")

Hello World
Welcome toScript Programming 101!


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [3]:
Age = 67
GPA = 3.67
Favorite_Color = "Black"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  67 <class 'int'>
GPA =  3.67 <class 'float'>
Favorite_Color =  Black <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [3]:
num1 = 20
num2 = 20
num3 = 20
num4 = 7
sum = num1 + num2 + num3 + num4
diff = num1 - num4
Prod = num1 * num2
Div = num1/num2

print("Sum = ",sum)
print("difference = ",diff)
print("Product = ",Prod)
print("Division = ",int(Div))

Sum =  67
difference =  13
Product =  400
Division =  1


4. String concatenation and f-strings:


In [6]:
First_name = "BAMBAM"
Last_name = "WILASINEE"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,BAMBAM WILASINEE


5. User input and type conversion:


In [4]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

type of A is :  <class 'str'>
62  + 5 =  67


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: Automating repetitive tasks

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: 13.5

ทำไมถึงเป็นtype float Because when an integer is added to a float in Python, the result is automatically converted to a float.

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old.

ทำไมถึงเป็นtype String Because when one string is added to another string in Python, the result is still a string.


In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: False

ทำไมถึงเป็นtype bool Because when one bool is added to another bool in Python, the result is still a bool.

In [9]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ได้ผลลัพธ์: no result

ทำไมถึงเป็น comment If you put a hash sign (#) at the beginning of a line in Python, the line becomes a comment.